### ISSUE TITLE
Bug report: cp_to_tensor with 1 dimensional factors fails

### ISSUE THREAD

**Issue by cohenjer** [tensorly/tensorly#409]
#### Describe the bug
When feeding one dimensional factors in `tl.cp_tensor.cp_to_tensor`, a shape mismatch error message is prompted even though there is no shape mismatch.

#### Steps or Code to Reproduce
```python
import tensorly as tl
import numpy as np

# This works
shapes = [(5,1),(2,1),(3,1)]
factors = [np.ones(shapes[0]),np.ones(shapes[1]),np.ones(shapes[2])]
tensor = tl.cp_tensor.cp_to_tensor((None,factors))
tensor = tl.cp_tensor.cp_to_tensor((np.array([1]),factors)) #another syntax
tensor = tl.tenalg.outer(factors)
# This does not work
shapes = [(5,),(2,),(3,)]
factors = [np.ones(shapes[0]),np.ones(shapes[1]),np.ones(shapes[2])]
tensor = tl.cp_tensor.cp_to_tensor((None,factors)) # shape mismatch error
tensor = tl.cp_tensor.cp_to_tensor((np.array([1]),factors)) # same error
tensor = tl.tenalg.outer(factors) # But we can use outer instead in this case
```

#### Expected behavior
I expected `cp_to_tensor` to work with 1d arrays for factors; it would be simple to support it by checking the dimensions of the factors in the pre-processing of `cp_to_tensor` and adding a mode in all factors.

**Comment by JeanKossaifi** [comment]
So I'm slightly conflicted with this one: it's an easy fix but one that adds yet more complexity in the reconstruction function, which is potentially called many times. 

Ideally I'd like to move to using more the factorized tensor classes so this can be tested once and the factors adjusted accordingly once, rather than checking every time.

**Comment by cohenjer** [comment]
I agree with you @JeanKossaifi; at the end of the day I wanted to signal that the error message is not very instructive. Updating the doc, and even the existence of this issue, might help users struggling like I was with the syntax here.
